src/transmission_electron.py

```diff
@@ -0,0 +1,274 @@
+        Returns
+        -------
+        classified_params : dict
+            Dictionary containing grouped parameters for TEM Simulator, with keys:
+                seed : str maps to int
+                    Seed for TEM Simulator
+                particle_mrcout : str maps to bool
+                    Flag for optional volume map of sample
+                sample_dimensions : str maps to
+                    List containing the specimen grid parameters
+                beam_params : str maps to list
+                    List containing the beam parameters
+                detector_params : str maps to list
+                    List containing the detector parameters
+                optics_params : str maps to list
+                    List containing the optic parameters
+        """
+        with open(config_yaml, "r") as stream:
+            raw_params = yaml.safe_load(stream)
+        classified_params = self.classify_input_config(raw_params)
+
+        return classified_params
+
+    @staticmethod
+    def classify_input_config(raw_params):
+        """Take dictionary of individual parameters and groups them into lists.
+
+        Parameters
+        ----------
+        raw_params : dict of type str to {str,bool,int}
+            Dictionary of simulator parameters
+        Returns
+        -------
+        classified_params : dict of type str to {str,bool,int,list}
+            Dictionary of grouped parameters
+        """
+        classified_params = {}
+        return classified_params
+
+    @staticmethod
+    def generate_file_paths(pdb_file, output_dir=None, mrc_keyword=None):
+        """Return the paths to pdb, crd, log, inp, and h5 files as strings.
+
+        Parameters
+        ----------
+        pdb_file : str
+            Relative path to the pdb file
+        output_dir : str, (default = None)
+            Relative path to output directory
+        mrc_keyword : str, (default = None)
+            user-specified keyword appended to output files
+        Returns
+        -------
+        file_paths : dict of type str to str
+            Dict of file paths that includes keys:
+            pdb_file
+                relative path to pdb input file
+            crd_file
+                relative path to desired output crd file
+            h5_file
+                relative path to desired output h5 file
+            inp_file
+                relative path to desired output inp file
+            mrc_file
+                relative path to desired output mrc file
+            log_file
+                relative path to desired output log file
+        """
+        file_paths = {}
+        return file_paths
+
+    @staticmethod
+    def create_crd_file(file_paths, sim_param_arrays, pad):
+        """Format and write molecular model data to crd_file for use in TEM-simulator.
+
+        Parameters
+        ----------
+        file_paths : dict of type str to str
```

Comment:
is this the same as `self.file_paths` from the constructor? if so, can we access it with `self.file_paths` directly?
same question for `sim_param_arrays` (which, in this case, should be called `self.sim_dict`?)